### PR TITLE
Install DeploySentinel Cypress debugger

### PR DESCRIPTION
**Pull request by wrn14897**
DeploySentinel Cypress debugger helps developers to debug flaky tests in CI

More details: https://www.npmjs.com/package/@deploysentinel/cypress-debugger

Quick demo:
https://user-images.githubusercontent.com/5959690/196567000-2aaf62df-eff6-409a-a160-db185750e473.mp4

### CODE REVIEW

File: .github/workflows/e2e-main.yml [hunk 1]
```diff
@@ -147,6 +147,8 @@ jobs:
           --ci-build-id "${{ github.run_id }}-${{ github.run_attempt }}"
       env:
         TERM: xterm
+        CYPRESS_DEPLOYSENTINEL_KEY: ${{ secrets.CYPRESS_DEPLOYSENTINEL_KEY }}
```
Comment (wrn14897) [inline .github/workflows/e2e-main.yml:150]: Need to grab api key from here https://www.deploysentinel.com/team

File: .github/workflows/e2e-main.yml [hunk 2]
```diff
@@ -147,6 +147,8 @@ jobs:
...
+        ELECTRON_EXTRA_LAUNCH_ARGS: '--remote-debugging-port=40500'
```
Comment (wrn14897) [inline .github/workflows/e2e-main.yml:151]: This config is for electron browser

File: frontend/test/__support__/e2e/cypress.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import "@deploysentinel/cypress-debugger/support";
```
Comment (wrn14897) [inline frontend/test/__support__/e2e/cypress.js:1]: Have this on the very top so debugger catch all exceptions

**Comment by nemanjaglumac** [comment]
Thank you @wrn14897 .
I've implemented it in https://github.com/metabase/metabase/pull/26036

File: frontend/test/__support__/e2e/config.js [hunk 4]
```diff
@@ -71,6 +71,8 @@ const defaultConfig = {
     config.env.SOURCE_VERSION = sourceVersion;
     config.env.TARGET_VERSION = targetVersion;
 
+    require("@deploysentinel/cypress-debugger/plugin")(on, config);
```
Comment (wrn14897) [inline frontend/test/__support__/e2e/config.js:74]: Need to put this after `before:browser:launch` custom listener to avoid conflict

**Pull request closed without merging**